syft/pkg/cataloger/binary/elf_package.go

```diff
@@ -16,7 +16,10 @@ func newELFPackage(metadata elfBinaryPackageNotes, locations file.LocationSet) p
 		PURL:      packageURL(metadata),
 		Type:      pkgType(metadata.Type),
 		Locations: locations,
-		Metadata:  metadata.ELFBinaryPackageNoteJSONPayload,
+		// though we can look for shared libs, we cannot see static dependencies nor dynamic dependencies using dlopen.
+		// this means that, even in cases where the dep info is actually complete, we can't programmatically determine that.
```

Comment:
I can't figure out whether this means it should be `Incomplete` or `Unknown`. Maybe add a sentence to the comment about why it's not `Unknown`?

Reply:
it should be unknown -- the table in the PR description is right, the code is wrong. I'll adjust the code.